echion/frame.h

```diff
@@ -385,8 +399,13 @@ Frame &Frame::read(PyObject *frame_addr, PyObject **prev_addr)
 
     // We cannot use _PyInterpreterFrame_LASTI because _PyCode_CODE reads
     // from the code object.
+#if PY_VERSION_HEX >= 0x030d0000
+    const int lasti = ((int)(iframe.instr_ptr - 1 - (_Py_CODEUNIT *)((PyCodeObject*)iframe.f_executable))) - offsetof(PyCodeObject, co_code_adaptive) / sizeof(_Py_CODEUNIT);
```

Comment:
From my understanding `instr_ptr` is either the currently executing instruction or the one about to execute, so subtracting one.